src/objects/tokens/mirage.js

```diff
@@ -0,0 +1,48 @@
+const { refObject, Rotator } = require("../../wrapper/api");
+const { System, Planet } = require("../../lib/system/system");
+
+const MIRAGE_ATTRS = {
+    localeName: "planet.mirage",
+    resources: 1,
+    influence: 2,
+    trait: ["cultural"],
+    position: { x: 2, y: -1.25 },
+    legendary: true,
+    legendaryCard: "card.legendary_planet:pok/mirage_flight_academy",
+};
+let MIRAGE_SYSTEM = 0;
+
+function attachMirage(obj) {
+    const pos = obj.getPosition();
+    const systemObj = System.getSystemTileObjectByPosition(pos);
+    if (systemObj) {
+        const system = System.getBySystemTileObject(systemObj);
+        if (MIRAGE_SYSTEM) {
```

Comment:
I'd move `removeExistingMirage` to a separate function.  Call it here like you are now, as well as in the Mirage GameObject.onGrab event handler (and attach in the onRelease handler).